src/machine/machine_stm32_gpio_reva.go

```diff
@@ -0,0 +1,85 @@
+// +build stm32,!stm32l4,!stm32l5
+
+package machine
+
+import (
+	"device/stm32"
+)
+
+//
+// STM32 allows one interrupt source per pin number, with
+// the same pin number in different ports sharing a single
+// interrupt source (so PA0, PB0, PC0 all share).  Only a
+// single physical pin can be connected to each pin number.
+//
+// To call interrupt callbacks, we record here for each
+// pin number the callback and the actual associated pin.
```

Comment:
Do I read this correctly that if you set an interrupt on `PA0`, and then one on `PB0`, that the interrupt on `PA0` is overwritten?

Reply:
Yes - there's physically a mux that selects one (and only one) of PA0, PB0, PC0, PD0, etc
![image](https://user-images.githubusercontent.com/509378/120423185-cb2b9b80-c31e-11eb-93ef-6cab0ee3a5cc.png)

Reply:
The API in all other implementations is that (from the API user perspective) every pin has a separate interrupt handler. With this implementation, setting an interrupt on pin PB0 would replace an interrupt on PA0, which is probably not what you'd expect given the current API.

Reply:
Also see this PR: https://github.com/tinygo-org/tinygo-site/pull/191

Reply:
With the latest changes, if you do something like this:
```go
machine.PA13.SetInterrupt(machine.PinChange,xxxx)
machine.PC13.SetInterrupt(machine.PinChange,xxxx)
```
You get an error: `could not configure pin interrupt: machine: no channel available for pin interrupt` on the second call to use slot 13.

Given the hardware constraints, it's probably the closest implementation to the interface?